### PR TITLE
Gl 753 spimm elgibility

### PR DESCRIPTION
**Pull request by shjohnson**
### Jira link

https://transformuk.atlassian.net/browse/GL-753

Adds in the SPIMM eligibility questions to be right after the start page

### CODE REVIEW

File: app/helpers/green_lanes_helper.rb [hunk 1]
```diff
@@ -54,6 +54,14 @@ def render_exemptions(assessments, category)
     end
   end
 
+  def yes_no_options
+    [OpenStruct.new(id: 'yes', name: 'Yes'), OpenStruct.new(id: 'no', name: 'No')]
```
Comment (alexdesi) [inline app/helpers/green_lanes_helper.rb:58]: nice one :rocket: 

I was just thinking that Struct may be enough, we don't need to expand them with new keys

File: app/helpers/green_lanes_helper.rb [hunk 2]
```diff
@@ -54,6 +54,14 @@ def render_exemptions(assessments, category)
...
+  end
+
+  def yes_no_not_sure_options
+    [OpenStruct.new(id: 'yes', name: 'Yes'), OpenStruct.new(id: 'no', name: 'No'), OpenStruct.new(id: 'not_sure', name: 'Not sure')]
```
Comment (alexdesi) [inline app/helpers/green_lanes_helper.rb:62]: same here
```
 [
  Struct.new(id: 'yes', name: 'Yes'), 
  Struct.new(id: 'no', name: 'No'),
  Struct.new(id: 'not_sure', name: 'Not sure')
]

```

File: app/services/green_lanes/eligibility_result.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
+module GreenLanes
+  class EligibilityResult
+    YOU_MAY_BE_ELIGIBLE = 'you_may_be_eligible'.freeze
+    NOT_YET_ELIGIBLE = 'not_yet_eligible'.freeze
+    NOT_ELIGIBLE = 'not_eligible'.freeze
+
+    NOT_SURE = 'not_sure'.freeze
+    YES = 'yes'.freeze
```
Comment (alexdesi) [inline app/services/green_lanes/eligibility_result.rb:8]: is `NO` missing?

File: spec/forms/green_lanes/eligibility_form_spec.rb [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+          free_circulation_in_uk: 'yes',
+          end_consumers_in_uk: 'yes',
+          ukims: 'yes',
+        }
+      end
+
+      it { expect(form.errors).to be_empty }
+    end
+
+    context 'when an commodity_code is missing' do
```
Comment (alexdesi) [inline spec/forms/green_lanes/eligibility_form_spec.rb]: `when a**n** commodity_code is missing`
or it should be
`when the commodity_code is missing`

File: spec/requests/green_lanes/eligibilities_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+      green_lanes_eligibility_form: {
+        moving_goods_gb_to_ni: 'yes',
+        free_circulation_in_uk: 'yes',
+        end_consumers_in_uk: 'yes',
+        ukims: 'yes',
+      },
+    }
+  end
+
+  let(:invalid_params) do
```
Comment (alexdesi) [inline spec/requests/green_lanes/eligibilities_controller_spec.rb]: we can move this inside the context
"with invalid params"

File: spec/requests/green_lanes/eligibilities_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,65 @@
+require 'spec_helper'
+
+RSpec.describe GreenLanes::EligibilitiesController, type: :request do
+  subject { make_request && response }
+
+  before do
+    allow(TradeTariffFrontend).to receive(:green_lanes_enabled?).and_return true
+  end
+
+  let(:valid_params) do
```
Comment (alexdesi) [inline spec/requests/green_lanes/eligibilities_controller_spec.rb]: maybe we should move this inside the appropriate context

File: app/services/green_lanes/eligibility_result.rb [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+      return NOT_YET_ELIGIBLE if not_yet_eligible?
+
+      NOT_ELIGIBLE
+    end
+
+    private
+
+    def not_eligible?
+      [@moving_goods_gb_to_ni, @free_circulation_in_uk, @end_consumers_in_uk].any?(&:!)
+    end
```
Comment (alexdesi) [inline app/services/green_lanes/eligibility_result.rb:28]: I'd suggest to use straigh logic:

```
def eligible?
  @moving_goods_gb_to_ni && @free_circulation_in_uk && @end_consumers_in_uk
end
```
and using `!eligible?` if we need the not_eligible

**Review by alexdesi — COMMENTED** [review]
It is very good,  I've only added minor comments

File: app/helpers/green_lanes_helper.rb (hunk 1 above)
Comment (shjohnson) [reply app/helpers/green_lanes_helper.rb:58]: Yeah possibly ive not tested to be honest, i cant remember why i used this pattern in the past

File: app/services/green_lanes/eligibility_result.rb (hunk 3 above)
Comment (shjohnson) [reply app/services/green_lanes/eligibility_result.rb:8]: i could add no your right and might make more sense for reading 👍

File: app/services/green_lanes/eligibility_result.rb (hunk 7 above)
Comment (shjohnson) [reply app/services/green_lanes/eligibility_result.rb:28]: I know weve discussed this in private so happy to change to 

`!@moving_goods_gb_to_ni || !@free_circulation_in_uk || !@end_consumers_in_uk` if preferred

File: spec/forms/green_lanes/eligibility_form_spec.rb (hunk 4 above)
Comment (shjohnson) [reply spec/forms/green_lanes/eligibility_form_spec.rb]: good catch and yes the bottom i think i prefer

File: spec/requests/green_lanes/eligibilities_controller_spec.rb (hunk 6 above)
Comment (shjohnson) [reply spec/requests/green_lanes/eligibilities_controller_spec.rb]: yeah i can have a look at this i think this is here how i first had the tests set up

File: app/services/green_lanes/eligibility_result.rb (hunk 3 above)
Comment (shjohnson) [reply app/services/green_lanes/eligibility_result.rb:8]: actually its not used so dont need this here to be honest..

File: app/helpers/green_lanes_helper.rb (hunk 1 above)
Comment (willfish) [reply app/helpers/green_lanes_helper.rb:58]: I think these relate more to the form than the green lanes helper?

File: app/services/green_lanes/eligibility_result.rb [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    def call
+      return YOU_MAY_BE_ELIGIBLE if may_be_eligible?
+      return NOT_YET_ELIGIBLE if not_yet_eligible?
+
+      NOT_ELIGIBLE
+    end
+
+    private
+
+    def not_eligible?
```
Comment (willfish) [inline app/services/green_lanes/eligibility_result.rb:26]: This isn't actually used anywhere

**Review by willfish — APPROVED** [review]

**Pull request merged**